clippy_lints/src/loops/unused_enumerate_index.rs

```diff
@@ -49,16 +39,16 @@ pub(super) fn check<'tcx>(cx: &LateContext<'tcx>, pat: &'tcx Pat<'_>, arg: &'tcx
     span_lint_and_then(
         cx,
         UNUSED_ENUMERATE_INDEX,
-        arg_span,
+        arg.span,
         "you seem to use `.enumerate()` and immediately discard the index",
         |diag| {
             let base_iter = sugg::Sugg::hir(cx, self_arg, "base iter");
             multispan_sugg(
                 diag,
                 "remove the `.enumerate()` call",
                 vec![
-                    (pat_span, snippet(cx, new_pat_span, "value").into_owned()),
-                    (arg_span, base_iter.to_string()),
+                    (pat.span, snippet(cx, tuple[1].span, "value").into_owned()),
```

Comment:
We usually use `..` as a snippet default.

```suggestion
                    (pat.span, snippet(cx, tuple[1].span, "..").into_owned()),
```